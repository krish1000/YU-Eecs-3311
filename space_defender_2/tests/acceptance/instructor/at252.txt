--Test enemy grunt armour/regen.
toggle_debug_mode

play(5,20,2,2,2,2,2)
setup_select(4)
setup_next(2)
setup_select(3)
setup_next(1)
setup_select(1)
setup_next(1)
setup_next(1)

move(B,2)
fire
move(B,1)
move(D,2)
move(D,1)
move(D,2)
move(D,1)
move(D,2)
