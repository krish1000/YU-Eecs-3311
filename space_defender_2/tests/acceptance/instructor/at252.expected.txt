  state:not started, normal, ok
  Welcome to Space Defender Version 2.
->toggle_debug_mode
  state:not started, debug, ok
  In debug mode.
->play(5,20,2,2,2,2,2)
  state:weapon setup, debug, ok
  1:Standard (A single projectile is fired in front)
    Health:10, Energy:10, Regen:0/1, Armour:0, Vision:1, Move:1, Move Cost:1,
    Projectile Damage:70, Projectile Cost:5 (energy)
  2:Spread (Three projectiles are fired in front, two going diagonal)
    Health:0, Energy:60, Regen:0/2, Armour:1, Vision:0, Move:0, Move Cost:2,
    Projectile Damage:50, Projectile Cost:10 (energy)
  3:Snipe (Fast and high damage projectile, but only travels via teleporting)
    Health:0, Energy:100, Regen:0/5, Armour:0, Vision:10, Move:3, Move Cost:0,
    Projectile Damage:1000, Projectile Cost:20 (energy)
  4:Rocket (Two projectiles appear behind to the sides of the Starfighter and accelerates)
    Health:10, Energy:0, Regen:10/0, Armour:2, Vision:2, Move:0, Move Cost:3,
    Projectile Damage:100, Projectile Cost:10 (health)
  5:Splitter (A single mine projectile is placed in front of the Starfighter)
    Health:0, Energy:100, Regen:0/10, Armour:0, Vision:0, Move:0, Move Cost:5,
    Projectile Damage:150, Projectile Cost:70 (energy)
  Weapon Selected:Standard
->setup_select(4)
  state:weapon setup, debug, ok
  1:Standard (A single projectile is fired in front)
    Health:10, Energy:10, Regen:0/1, Armour:0, Vision:1, Move:1, Move Cost:1,
    Projectile Damage:70, Projectile Cost:5 (energy)
  2:Spread (Three projectiles are fired in front, two going diagonal)
    Health:0, Energy:60, Regen:0/2, Armour:1, Vision:0, Move:0, Move Cost:2,
    Projectile Damage:50, Projectile Cost:10 (energy)
  3:Snipe (Fast and high damage projectile, but only travels via teleporting)
    Health:0, Energy:100, Regen:0/5, Armour:0, Vision:10, Move:3, Move Cost:0,
    Projectile Damage:1000, Projectile Cost:20 (energy)
  4:Rocket (Two projectiles appear behind to the sides of the Starfighter and accelerates)
    Health:10, Energy:0, Regen:10/0, Armour:2, Vision:2, Move:0, Move Cost:3,
    Projectile Damage:100, Projectile Cost:10 (health)
  5:Splitter (A single mine projectile is placed in front of the Starfighter)
    Health:0, Energy:100, Regen:0/10, Armour:0, Vision:0, Move:0, Move Cost:5,
    Projectile Damage:150, Projectile Cost:70 (energy)
  Weapon Selected:Rocket
->setup_next(2)
  state:engine setup, debug, ok
  1:Standard
    Health:10, Energy:60, Regen:0/2, Armour:1, Vision:12, Move:8, Move Cost:2
  2:Light
    Health:0, Energy:30, Regen:0/1, Armour:0, Vision:15, Move:10, Move Cost:1
  3:Armoured
    Health:50, Energy:100, Regen:0/3, Armour:3, Vision:6, Move:4, Move Cost:5
  Engine Selected:Standard
->setup_select(3)
  state:engine setup, debug, ok
  1:Standard
    Health:10, Energy:60, Regen:0/2, Armour:1, Vision:12, Move:8, Move Cost:2
  2:Light
    Health:0, Energy:30, Regen:0/1, Armour:0, Vision:15, Move:10, Move Cost:1
  3:Armoured
    Health:50, Energy:100, Regen:0/3, Armour:3, Vision:6, Move:4, Move Cost:5
  Engine Selected:Armoured
->setup_next(1)
  state:power setup, debug, ok
  1:Recall (50 energy): Teleport back to spawn.
  2:Repair (50 energy): Gain 50 health, can go over max health. Health regen will not be in effect if over cap.
  3:Overcharge (up to 50 health): Gain 2*health spent energy, can go over max energy. Energy regen will not be in effect if over cap.
  4:Deploy Drones (100 energy): Clear all projectiles.
  5:Orbital Strike (100 energy): Deal 100 damage to all enemies, affected by armour.
  Power Selected:Recall (50 energy): Teleport back to spawn.
->setup_select(1)
  state:power setup, debug, ok
  1:Recall (50 energy): Teleport back to spawn.
  2:Repair (50 energy): Gain 50 health, can go over max health. Health regen will not be in effect if over cap.
  3:Overcharge (up to 50 health): Gain 2*health spent energy, can go over max energy. Energy regen will not be in effect if over cap.
  4:Deploy Drones (100 energy): Clear all projectiles.
  5:Orbital Strike (100 energy): Deal 100 damage to all enemies, affected by armour.
  Power Selected:Recall (50 energy): Teleport back to spawn.
->setup_next(1)
  state:setup summary, debug, ok
  Weapon Selected:Rocket
  Armour Selected:None
  Engine Selected:Armoured
  Power Selected:Recall (50 energy): Teleport back to spawn.
->setup_next(1)
  state:in game(0.0), debug, ok
  Starfighter:
    [0,S]->health:110/110, energy:100/100, Regen:11/3, Armour:5, Vision:8, Move:5, Move Cost:8, location:[C,1]
      Projectile Pattern:Rocket, Projectile Damage:100, Projectile Cost:10 (health)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
  Projectile:
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
  Enemy Action:
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20
    A _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C S  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->move(B,2)
  state:in game(1.0), debug, ok
  Starfighter:
    [0,S]->health:110/110, energy:84/100, Regen:11/3, Armour:5, Vision:8, Move:5, Move Cost:8, location:[B,2]
      Projectile Pattern:Rocket, Projectile Damage:100, Projectile Cost:10 (health)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,G]->health:100/100, Regen:1, Armour:1, Vision:5, seen_by_Starfighter:F, can_see_Starfighter:F, location:[A,20]
  Projectile:
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) moves: [C,1] -> [B,2]
  Enemy Action:
  Natural Enemy Spawn:
    A Grunt(id:1) spawns at location [A,20].
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20
    A _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  G
    B _  S  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->fire
  state:in game(2.0), debug, ok
  Starfighter:
    [0,S]->health:100/110, energy:87/100, Regen:11/3, Armour:5, Vision:8, Move:5, Move Cost:8, location:[B,2]
      Projectile Pattern:Rocket, Projectile Damage:100, Projectile Cost:10 (health)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,G]->health:100/100, Regen:1, Armour:1, Vision:5, seen_by_Starfighter:F, can_see_Starfighter:F, location:[A,18]
  Projectile:
    [-1,*]->damage:100, move:1, location:[A,1]
    [-2,*]->damage:100, move:1, location:[C,1]
    [-3,<]->damage:15, move:4, location:[A,17]
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) fires at location [B,2].
      A friendly projectile(id:-1) spawns at location [A,1].
      A friendly projectile(id:-2) spawns at location [C,1].
  Enemy Action:
    A Grunt(id:1) moves: [A,20] -> [A,18]
      A enemy projectile(id:-3) spawns at location [A,17].
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20
    A *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  <  G  _  _
    B _  S  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->move(B,1)
  state:in game(3.0), debug, ok
  Starfighter:
    [0,S]->health:110/110, energy:82/100, Regen:11/3, Armour:5, Vision:8, Move:5, Move Cost:8, location:[B,1]
      Projectile Pattern:Rocket, Projectile Damage:100, Projectile Cost:10 (health)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,G]->health:100/100, Regen:1, Armour:1, Vision:5, seen_by_Starfighter:F, can_see_Starfighter:F, location:[A,16]
  Projectile:
    [-1,*]->damage:100, move:2, location:[A,2]
    [-2,*]->damage:100, move:2, location:[C,2]
    [-3,<]->damage:15, move:4, location:[A,13]
    [-4,<]->damage:15, move:4, location:[A,15]
  Friendly Projectile Action:
    A friendly projectile(id:-1) moves: [A,1] -> [A,2]
    A friendly projectile(id:-2) moves: [C,1] -> [C,2]
  Enemy Projectile Action:
    A enemy projectile(id:-3) moves: [A,17] -> [A,13]
  Starfighter Action:
    The Starfighter(id:0) moves: [B,2] -> [B,1]
  Enemy Action:
    A Grunt(id:1) moves: [A,18] -> [A,16]
      A enemy projectile(id:-4) spawns at location [A,15].
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20
    A _  *  _  _  _  _  _  _  _  _  _  _  <  _  <  G  _  _  _  _
    B S  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->move(D,2)
  state:in game(4.0), debug, ok
  Starfighter:
    [0,S]->health:110/110, energy:61/100, Regen:11/3, Armour:5, Vision:8, Move:5, Move Cost:8, location:[D,2]
      Projectile Pattern:Rocket, Projectile Damage:100, Projectile Cost:10 (health)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,G]->health:100/100, Regen:1, Armour:1, Vision:5, seen_by_Starfighter:F, can_see_Starfighter:F, location:[A,14]
  Projectile:
    [-1,*]->damage:100, move:4, location:[A,4]
    [-2,*]->damage:100, move:4, location:[C,4]
    [-3,<]->damage:15, move:4, location:[A,9]
    [-4,<]->damage:15, move:4, location:[A,11]
    [-5,<]->damage:15, move:4, location:[A,13]
  Friendly Projectile Action:
    A friendly projectile(id:-1) moves: [A,2] -> [A,4]
    A friendly projectile(id:-2) moves: [C,2] -> [C,4]
  Enemy Projectile Action:
    A enemy projectile(id:-3) moves: [A,13] -> [A,9]
    A enemy projectile(id:-4) moves: [A,15] -> [A,11]
  Starfighter Action:
    The Starfighter(id:0) moves: [B,1] -> [D,2]
  Enemy Action:
    A Grunt(id:1) moves: [A,16] -> [A,14]
      A enemy projectile(id:-5) spawns at location [A,13].
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20
    A _  _  _  *  _  _  _  _  <  _  <  _  <  G  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  S  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->move(D,1)
  state:in game(5.0), debug, ok
  Starfighter:
    [0,S]->health:110/110, energy:56/100, Regen:11/3, Armour:5, Vision:8, Move:5, Move Cost:8, location:[D,1]
      Projectile Pattern:Rocket, Projectile Damage:100, Projectile Cost:10 (health)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,G]->health:100/100, Regen:1, Armour:1, Vision:5, seen_by_Starfighter:F, can_see_Starfighter:F, location:[A,12]
  Projectile:
    [-1,*]->damage:70, move:8, location:[A,8]
    [-2,*]->damage:100, move:8, location:[C,8]
    [-5,<]->damage:15, move:4, location:[A,9]
    [-6,<]->damage:15, move:4, location:[A,11]
  Friendly Projectile Action:
    A friendly projectile(id:-1) moves: [A,4] -> [A,8]
    A friendly projectile(id:-2) moves: [C,4] -> [C,8]
  Enemy Projectile Action:
    A enemy projectile(id:-3) moves: [A,9] -> [A,8]
      The projectile collides with friendly projectile(id:-1) at location [A,8], negating damage.
    A enemy projectile(id:-4) moves: [A,11] -> [A,8]
      The projectile collides with friendly projectile(id:-1) at location [A,8], negating damage.
    A enemy projectile(id:-5) moves: [A,13] -> [A,9]
  Starfighter Action:
    The Starfighter(id:0) moves: [D,2] -> [D,1]
  Enemy Action:
    A Grunt(id:1) moves: [A,14] -> [A,12]
      A enemy projectile(id:-6) spawns at location [A,11].
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20
    A _  _  _  _  _  _  _  *  <  _  <  G  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  *  _  _  _  _  _  _  _  _  _  _  _  _
    D S  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->move(D,2)
  state:in game(6.0), debug, ok
  Starfighter:
    [0,S]->health:110/110, energy:51/100, Regen:11/3, Armour:5, Vision:8, Move:5, Move Cost:8, location:[D,2]
      Projectile Pattern:Rocket, Projectile Damage:100, Projectile Cost:10 (health)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,G]->health:62/100, Regen:1, Armour:1, Vision:5, seen_by_Starfighter:F, can_see_Starfighter:F, location:[A,10]
  Projectile:
    [-2,*]->damage:100, move:16, location:[C,16]
    [-7,<]->damage:15, move:4, location:[A,9]
  Friendly Projectile Action:
    A friendly projectile(id:-1) moves: [A,8] -> [A,12]
      The projectile collides with enemy projectile(id:-5) at location [A,9], negating damage.
      The projectile collides with enemy projectile(id:-6) at location [A,11], negating damage.
      The projectile collides with Grunt(id:1) at location [A,12], dealing 39 damage.
    A friendly projectile(id:-2) moves: [C,8] -> [C,16]
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) moves: [D,1] -> [D,2]
  Enemy Action:
    A Grunt(id:1) moves: [A,12] -> [A,10]
      A enemy projectile(id:-7) spawns at location [A,9].
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20
    A _  _  _  _  _  _  _  _  <  G  _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  *  _  _  _  _
    D _  S  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->move(D,1)
  state:in game(7.0), debug, ok
  Starfighter:
    [0,S]->health:110/110, energy:46/100, Regen:11/3, Armour:5, Vision:8, Move:5, Move Cost:8, location:[D,1]
      Projectile Pattern:Rocket, Projectile Damage:100, Projectile Cost:10 (health)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,G]->health:63/100, Regen:1, Armour:1, Vision:5, seen_by_Starfighter:F, can_see_Starfighter:F, location:[A,8]
  Projectile:
    [-7,<]->damage:15, move:4, location:[A,5]
    [-8,<]->damage:15, move:4, location:[A,7]
  Friendly Projectile Action:
    A friendly projectile(id:-2) moves: [C,16] -> out of board
  Enemy Projectile Action:
    A enemy projectile(id:-7) moves: [A,9] -> [A,5]
  Starfighter Action:
    The Starfighter(id:0) moves: [D,2] -> [D,1]
  Enemy Action:
    A Grunt(id:1) moves: [A,10] -> [A,8]
      A enemy projectile(id:-8) spawns at location [A,7].
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20
    A _  _  _  _  <  _  <  G  _  _  _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D S  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->move(D,2)
  state:in game(8.0), debug, ok
  Starfighter:
    [0,S]->health:110/110, energy:41/100, Regen:11/3, Armour:5, Vision:8, Move:5, Move Cost:8, location:[D,2]
      Projectile Pattern:Rocket, Projectile Damage:100, Projectile Cost:10 (health)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,G]->health:64/100, Regen:1, Armour:1, Vision:5, seen_by_Starfighter:T, can_see_Starfighter:F, location:[A,6]
  Projectile:
    [-7,<]->damage:15, move:4, location:[A,1]
    [-8,<]->damage:15, move:4, location:[A,3]
    [-9,<]->damage:15, move:4, location:[A,5]
  Friendly Projectile Action:
  Enemy Projectile Action:
    A enemy projectile(id:-7) moves: [A,5] -> [A,1]
    A enemy projectile(id:-8) moves: [A,7] -> [A,3]
  Starfighter Action:
    The Starfighter(id:0) moves: [D,1] -> [D,2]
  Enemy Action:
    A Grunt(id:1) moves: [A,8] -> [A,6]
      A enemy projectile(id:-9) spawns at location [A,5].
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20
    A <  _  <  _  <  G  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  S  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
